src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -51,7 +51,7 @@ public static void AnalyzeCompilation(CompilationAnalysisContext context)
 
             private static bool IsSourceLocation(Location location)
             {
-                return location != null && location.Kind == LocationKind.SourceFile;
+                return location is { Kind: LocationKind.SourceFile };
```

Comment:
>location [](start = 23, length = 8)

Nice.

I'm going to stop here as I think you have enough feedback to make a couple of substantive improvements.

Reply:
Thank you so much!!